src/sentry/incidents/endpoints/serializers/workflow_engine_incident.py

```diff
@@ -30,71 +49,200 @@ class WorkflowEngineIncidentSerializer(Serializer):
-        self, open_period: GroupOpenPeriod
-    ) -> list[IncidentActivitySerializerResponse]:
-        # TODO: Implement this
-        return [
-            {
-                "id": "-1",
-                "incidentIdentifier": "-1",
-                "user": {},
-                "type": 1,
-                "value": "test",
-                "previousValue": "test",
-                "comment": "test",
-                "dateCreated": datetime.now(),
-            }
-        ]
-
-    def get_alert_rule(self, open_period: GroupOpenPeriod) -> AlertRuleSerializerResponse:
-        # TODO: Implement this
-        return {
-            "id": "-1",
-            "name": "Test Alert Rule",
-            "organizationId": "-1",
-            "status": 1,
-            "query": "test",
-            "aggregate": "test",
-            "timeWindow": 1,
-            "resolution": 1,
-            "thresholdPeriod": 1,
-            "triggers": [
-                {
-                    "id": "-1",
-                    "status": 1,
-                    "dateModified": datetime.now(),
-                    "dateCreated": datetime.now(),
-                }
-            ],
-            "dateModified": datetime.now(),
-            "dateCreated": datetime.now(),
-            "createdBy": {},
-            "description": "test",
-            "detectionType": "test",
+    def get_open_period_activities(self, open_period: GroupOpenPeriod) -> list[dict[str, Any]]:
+        # XXX: an incident will be 1:1 with open periods, but there can be multiple open periods per metric issue
+        # XXX: this won't actually work until we start writing to the table for metric issues (or are we planning a backfill?)
+
+        open_period_activities = []
+        incident_activity_id = "-1"  # temp until we add lookup table
+        incident_identifier = "-1"  # temp until we add a column
+
+        # if we are here we have both IncidentActivityType.CREATED and IncidentActivityType.DETECTED
+        created = {
+            "id": incident_activity_id,
+            "incidentIdentifier": incident_identifier,
+            "type": IncidentActivityType.CREATED,
+            "value": None,
+            "previousValue": None,
+            "user": None,
+            "comment": None,
+            "dateCreated": open_period.date_started,
         }
+        detected = created.copy()
+        detected["type"] = IncidentActivityType.DETECTED
+        open_period_activities.append(created)
+        open_period_activities.append(detected)
+
+        # look up Activity rows for other status changes (warning, critical, and resolved)
+        activity_status_to_incident_status: dict[str, Any] = {
+            "high": IncidentStatus.CRITICAL,
+            "medium": IncidentStatus.WARNING,
+        }
+        status_change_activities = Activity.objects.filter(
+            group=open_period.group,
+            type__in=[ActivityType.SET_PRIORITY.value, ActivityType.SET_RESOLVED.value],
```

Comment:
realized this with the open period backfills - we probably want to look for all the `SET_RESOLVED_XXX` activity types for resolved in release/commit/etc